tests/system/aiplatform/test_dataset.py

```diff
@@ -350,3 +352,27 @@ def test_export_data(self, storage_client, staging_bucket):
         blob = bucket.get_blob(prefix)
 
         assert blob  # Verify the returned GCS export path exists
+
+    def test_update_dataset(self):
+        """Create a new dataset and use update() method to change its display_name, labels, and description.
+        Then confirm these fields of the dataset was successfully modifed."""
+
+        try:
+            dataset = aiplatform.ImageDataset.create()
+            labels = dataset.labels
+
+            dataset = dataset.update(
+                display_name=_TEST_DATASET_DISPLAY_NAME,
+                labels=_TEST_DATASET_LABELS,
+                description=_TEST_DATASET_DESCRIPTION,
+                update_request_timeout=None,
+            )
+            labels.update(_TEST_DATASET_LABELS)
+
+            assert dataset.display_name == _TEST_DATASET_DISPLAY_NAME
+            assert dataset.labels == labels
+            assert dataset.gca_resource.description == _TEST_DATASET_DESCRIPTION
+
+        finally:
+            if dataset is not None:
```

Comment:
This line is superfluous. `dataset` will only exist on assignment after `ImageDataset.create` is successful.

The test should start with `dataset=None`

Reply:
If the `ImageDataset.create` is successful, then we want to delete the dataset finally. But if `create()` fails, dataset is None and we shouldn't call `dataset.delete()`. So I think it's necessary to add this line.

Reply:
If the `ImageDataset.create` is successful, then we want to delete the dataset finally. But if `create()` fails, dataset is None and we shouldn't call `dataset.delete()`. So I think it's necessary to add this line.

Reply:
If create fails then `dataset`  is not defined and `if dataset is not None` will throw a `NameError`. `dataset` isn't created until after the right side of the statement, ` dataset = aiplatform.ImageDataset.create()` is successful. You can validate this behavior in Python:

```Python
def f():
    raise RuntimeError('')
    
try:
    dataset = f()
except:
    if dataset is None:
        pass
```

Reply:
Got it! 

But I see other system tests like `test_create_and_import_image_dataset` and `test_create_tabular_dataset` have the same pattern that checks `if dataset is not None`. Is there any reason for that?

Reply:
No, those should be eventually fixed as well.

Reply:
Got it. Just removed this line. Thank you!